notifications_utils/recipients.py

```diff
@@ -386,26 +386,32 @@ def validate_local_phone_number(number, column=None):
         raise InvalidPhoneError('Not a valid local number')
 
 
-def validate_phone_number(number, column=None, international=False):
+def validate_phone_number(number, column=None, international=False):  # noqa:   C901
 
     if ';' in number:
         raise InvalidPhoneError('Not a valid number')
 
     if (not international) or is_local_phone_number(number):
         return validate_local_phone_number(number)
 
-    number = normalise_phone_number(number)
+    normalized_number = normalise_phone_number(number)
 
-    if number is False:
-        raise InvalidPhoneError('Not a valid international number')
-
-    if len(number) < 8:
+    if normalized_number is False:
+        try:
+            parsedNumber = phonenumbers.parse(number, region_code)
+            isValidNumber = phonenumbers.is_valid_number(parsedNumber)
+            if (isValidNumber is False):
+                raise InvalidPhoneError("Field contains an invalid number due to either formatting or an impossible combination of area code and/or telephone prefix.")
+        except phonenumbers.phonenumberutil.NumberParseException:
+            raise InvalidPhoneError('Not a valid international number')
+
+    if len(normalized_number) < 8:
```

Comment:
Is this ever going to be false? How would it parse the number, check for a valid number, and then still not be the correct length?

Reply:
this condition was already existing and the false means that the number is not parsed. So if number is valid, parsed, normalized, it will be a string.